src/ucp/core/ucp_context.h

```diff
@@ -643,10 +645,28 @@ int ucp_is_scalable_transport(ucp_context_h context, size_t max_num_eps)
     return (max_num_eps >= (size_t)context->config.est_num_eps);
 }
 
+static UCS_F_ALWAYS_INLINE double
+ucp_context_get_est_num_eps(ucp_context_h context, int is_prioritized)
+{
+    return is_prioritized ? ucs_min(context->config.est_num_eps,
+                                    context->config.ext.prio_max_num_eps) :
+                            context->config.est_num_eps;
+}
+
+static UCS_F_ALWAYS_INLINE double
+ucp_tl_iface_latency_with_priority(ucp_context_h context,
+                                   const ucs_linear_func_t *latency,
+                                   int is_prioritized)
+{
+    unsigned num_eps = ucp_context_get_est_num_eps(context, is_prioritized);
+
+    return ucs_linear_func_apply(*latency, num_eps);
```

Comment:
as we discussed offline, if context->config.est_num_eps is not set, and is_prioritized==0, and latency->m>0, we should return INF latency to prevent RC from being selected on large scale for non-prioritized endpoint